File: 125_valid_palindrome.js
var isPalindrome = function(s) {
    s = s.replace(/[^0-9a-z]/gi, '').toLowerCase();

    if (s.length < 2){
        return true;
    }

    let left = 0;
    let right = s.length - 1;
    while (left <= right){
        if (s[left] !== s[right]){
            return false;
        } else {
            left++;
            right--;
        }
    }
    return true;
    //Time Complexity O(n), Space Complexity O(1)
};

File: 125_valid_palindrome.py
import re

class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = re.sub(r'[^0-9a-z]', '', s, flags=re.IGNORECASE)
        s = s.lower()

        if len(s) < 2:
            return True
        
        left = 0
        right = len(s) - 1

        while left <= right:
            if s[left] != s[right]:
                return False
            else:
                left+=1
                right-=1
        return True